cura/Settings/MaterialSettingsVisibilityHandler.py

```diff
@@ -0,0 +1,19 @@
+# Copyright (c) 2016 Ultimaker B.V.
+# Uranium is released under the terms of the AGPLv3 or higher.
+
+import UM.Settings.Models
+
+class MaterialSettingsVisibilityHandler(UM.Settings.Models.SettingVisibilityHandler):
+    def __init__(self, parent = None, *args, **kwargs):
+        super().__init__(parent = parent, *args, **kwargs)
+
+        material_settings = set([
+            "material_print_temperature",
+            "material_bed_temperature",
+            "material_standby_temperature",
+            "cool_fan_speed",
+            "retraction_amount",
+            "retraction_speed",
+        ])
```

Comment:
This is a visibility handler. It specifies which settings should be visible through SettingDefinitionsModel. In this case, the list of settings supported by the XML profiles is a fixed, limited list so there is no reason to use a preference. If anything, the dictionary in XmlMaterialProfile should be re-used for this, but that would create a tricky dependency on a plugin.

The fact that CuraApplication speficies a certain default set of visible settings is rather unrelated to this. There is no duplication of data with the list of visible settings there since they are used for different purposes.